### PR TITLE
Chi 1661 caselist date filter error

### PR DESCRIPTION
**Pull request by acedeywin**
Primary reviewer: @stephenhand 

## Description

Bug fix - fix case list date filter error. While investigating the fix to this issue, I found that we can only use onClick to perform event actions rather than onChange while filtering form date.
### Checklist
- [x] Corresponding issue has been opened
- [N/A] New tests added
- [N/A] Feature flags added
- [N/A] Strings are localized
- [N/A] Tested for chat contacts
- [N/A] Tested for call contacts

### Related Issues
Fixes CHI-1661

### CODE REVIEW

File: plugin-hrm-form/src/components/caseList/filters/DateRangeFilter.tsx [hunk 1]
```diff
@@ -74,7 +75,8 @@ const formToDateFilter = (
   filterOptions: DateFilterOption[],
   values: ReactHookFormValues,
 ): DateFilterValue | undefined => {
-  const [, selected] = filterOptions.find(([opt]) => opt === values[selectedOptionField]);
+  const { ...selected } = Object(filterOptions.find(([opt]) => opt === values[selectedOptionField]))[1];
```
Comment (stephenhand) [inline plugin-hrm-form/src/components/caseList/filters/DateRangeFilter.tsx:78]: This works but is a little messy and hard to follow, all we really need to do to handle the undefined cases is add some defensive checks:

````ts
const [, selected] = filterOptions?.find(([opt]) => opt === values[selectedOptionField]) ?? []
````

Comment (acedeywin) [reply plugin-hrm-form/src/components/caseList/filters/DateRangeFilter.tsx:78]: @stephenhand, you are looking at the wrong PR. [Here](https://github.com/techmatters/flex-plugins/pull/1228) is the updated PR. Your suggested implementation works perfectly.
